app/Models/V2/ScheduledJobs/TaskDueJob.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace App\Models\V2\ScheduledJobs;
+
+use App\Models\V2\Action;
+use App\Models\V2\Projects\Project;
+use App\Models\V2\Projects\ProjectReport;
+use App\Models\V2\Tasks\Task;
+use App\StateMachines\ReportStatusStateMachine;
+use App\StateMachines\TaskStatusStateMachine;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Builder;
+use Parental\HasParent;
+
+/**
+ * @property string $framework_key
+ * @property Carbon $due_at
+ * @property string $period_key
+ */
+class TaskDueJob extends ScheduledJob
+{
+    use HasParent;
+
+    public static function createTaskDue(Carbon $executionTime, string $frameworkKey, Carbon $dueAt): TaskDueJob
+    {
+        return self::create([
+            'execution_time' => $executionTime,
+            'task_definition' => [
+                'framework_key' => $frameworkKey,
+                'due_at' => $dueAt,
+            ],
+        ]);
+    }
+
+    public function scopeFramework(Builder $query, string $frameworkKey): Builder
+    {
+        return $query->whereJsonContains('task_definition->framework_key', $frameworkKey)
+            ->orderBy('execution_time');
+    }
```

Comment:
This scope makes it easy to determine the next time a given framework is set to either generate reports, or when those reports will be due (in conjunction with the `getDueAtAttribute` below):
```
> TaskDueJob::framework('terrafund')->first()->execution_time
= Illuminate\Support\Carbon @1735603200 {#9010
    date: 2024-12-31 00:00:00.0 UTC (+00:00),
  }

> TaskDueJob::framework('terrafund')->first()->due_at
= Carbon\Carbon @1738281600 {#9030
    date: 2025-01-31 00:00:00.0 +00:00,
  }
```